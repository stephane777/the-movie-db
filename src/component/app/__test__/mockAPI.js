export const resultMockAPI = [
	{
		adult: false,
		backdrop_path: "/jMWkd0fuwbG39eJpzycJzPWMCww.jpg",
		genre_ids: [28, 878],
		id: 399566,
		original_language: "en",
		original_title: "Godzilla vs. Kong",
		overview:
			"In a time when monsters walk the Earth, humanity’s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.",
		popularity: 12276.164,
		poster_path: "/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg",
		release_date: "2021-03-24",
		title: "Godzilla vs. Kong",
		video: false,
		vote_average: 8.5,
		vote_count: 3719,
	},
	{
		adult: false,
		backdrop_path: "/hJuDvwzS0SPlsE6MNFOpznQltDZ.jpg",
		genre_ids: [16, 12, 14, 10751, 28],
		id: 527774,
		original_language: "en",
		original_title: "Raya and the Last Dragon",
		overview:
			"Long ago, in the fantasy world of Kumandra, humans and dragons lived together in harmony. But when an evil force threatened the land, the dragons sacrificed themselves to save humanity. Now, 500 years later, that same evil has returned and it’s up to a lone warrior, Raya, to track down the legendary last dragon to restore the fractured land and its divided people.",
		popularity: 2495.914,
		poster_path: "/lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg",
		release_date: "2021-03-03",
		title: "Raya and the Last Dragon",
		video: false,
		vote_average: 8.3,
		vote_count: 2088,
	},
	{
		adult: false,
		backdrop_path: "/fev8UFNFFYsD5q7AcYS8LyTzqwl.jpg",
		genre_ids: [35, 10751, 16],
		id: 587807,
		original_language: "en",
		original_title: "Tom & Jerry",
		overview:
			"Tom the cat and Jerry the mouse get kicked out of their home and relocate to a fancy New York hotel, where a scrappy employee named Kayla will lose her job if she can’t evict Jerry before a high-class wedding at the hotel. Her solution? Hiring Tom to get rid of the pesky mouse.",
		popularity: 1689.341,
		poster_path: "/6KErczPBROQty7QoIsaa6wJYXZi.jpg",
		release_date: "2021-02-11",
		title: "Tom & Jerry",
		video: false,
		vote_average: 7.4,
		vote_count: 1130,
	},
];
