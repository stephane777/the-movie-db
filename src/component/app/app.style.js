import styled from "styled-components";
import Button from "react-bootstrap/Button";
import Row from "react-bootstrap/Row";

export const Heading = styled.h2`
	font-size: 2.6rem;
	width: 100%;
	text-align: left;
	padding-left: 4rem;
	font-weight: bold;
	font-family: "Source Sans Pro", Arial, sans-serif;
	@media only screen and (max-width: 768px) {
		text-align: center;
	}
`;

export const SearchButton = styled(Button)`
	background-color: ${({ disabled }) =>
		disabled
			? "var(--lightGrey) !important"
			: "var(--tmdbLightBlue) !important"};
	// min-width: 28rem;
	border-color: ${({ disabled }) =>
		disabled
			? "var(--lightGrey) !important"
			: "var(--tmdbLightBlue) !important"};
	color: ${({ disabled }) =>
		disabled ? "#000 !important" : "#fff !important"};
	max-width: 46rem;
	width: 100%;
	font-size: 1.6rem;
	padding: 0.8rem 0;
	border-radius: 20px;
	font-size: 1.8rem;
	font-weight: bold;
	&:hover {
		background-color: var(--tmdbLightBlue);
	}
`;

export const RespRow = styled(Row)`
	@media only screen and (max-width: 768px) {
		justify-content: center !important;
	}
`;
