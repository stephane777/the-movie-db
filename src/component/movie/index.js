export { Movie } from "./movie.component";
