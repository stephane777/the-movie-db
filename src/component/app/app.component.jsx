import React from "react";
import axios from "axios";
import { Container, Row, Col } from "react-bootstrap";
import { v4 as uuidv4 } from "uuid";
import { Movie } from "../movie";
import { Filter } from "../filter";
import { Banner } from "../banner";
import { Footer } from "../footer";
import { reducer } from "../../reducer";
import { Loading } from "../loading";

import * as SA from "./app.style";

const initialState = {
	data: [],
	error: "",
	loading: false,
	sortRating: "Descending",
	newFilterSelected: "Descending",
	chevron: false,
};
const App = () => {
	const [state, dispatch] = React.useReducer(reducer, initialState);
	/**
	 * state.chevron : true means Sort is expanded, flase collapsed
	 * state.newFilterSelected : help to compare if different from sortRating and handle Search btn status
	 * state.sortRating : current filter applied in the UI
	 */

	const {
		data,
		loading,
		error,
		sortRating,
		newFilterSelected,
		chevron,
	} = state;

	React.useEffect(() => {
		const getMovies = async () => {
			try {
				dispatch({ type: "LOADING", status: true });
				const api_key = process.env.REACT_APP_API_KEY;
				const data = await axios.get(
					`https://api.themoviedb.org/3/movie/now_playing`,
					{
						params: {
							api_key,
							language: "en_US",
							page: 1,
						},
					}
				);
				const {
					data: { results },
				} = await data;
				const resultSorted = results
					.sort((a, b) => a.vote_average - b.vote_average)
					.reverse();

				dispatch({ type: "SET_DATA", data: resultSorted });
				dispatch({ type: "LOADING", status: false });
			} catch (e) {
				dispatch({ type: "LOADING", status: false });
				dispatch({ type: "ERROR", message: e.toString() });
			}
		};
		getMovies();
	}, []);

	const handleNewFilterSelected = (e) => {
		console.log(`e.target.value: ${e.target.value}`);
		dispatch({ type: "NEW_FILTER_SELECTED", direction: e.target.value });
	};
	const handleSortRating = (e) => {
		dispatch({ type: "SORT_RATING", direction: newFilterSelected });
		dispatch({ type: "SORT_DATA", direction: sortRating });
	};

	const wrapper = (
		<React.Fragment>
			<Container style={{ maxWidth: "1600px" }} className="px-0">
				<Row className="my-5 mx-0">
					<SA.Heading>Now Playing Movies</SA.Heading>
				</Row>
				<Row className="d-flex justify-content-center mx-0">
					<Col md="4" className="d-flex flex-column align-items-center">
						<Filter
							chevron={chevron}
							setChevron={() => dispatch({ type: "SET_CHEVRON" })}
							handleNewFilterSelected={handleNewFilterSelected}
							newFilterSelected={newFilterSelected}
						/>
						<SA.SearchButton
							onClick={handleSortRating}
							disabled={sortRating === newFilterSelected}
						>
							Search
						</SA.SearchButton>
					</Col>
					<Col md="8">
						<SA.RespRow className="justify-content-start">
							{data.map((mov, i) => (
								<Movie key={uuidv4()} data={mov} />
							))}
						</SA.RespRow>
					</Col>
				</Row>
			</Container>
		</React.Fragment>
	);
	return (
		<div className="App">
			<div>
				<Banner />

				{loading && <Loading />}
				{error && <p>{error}</p>}
				{!error && !loading && wrapper}
				<Footer />
			</div>
		</div>
	);
};

export { App };
