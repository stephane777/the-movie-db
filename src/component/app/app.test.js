import React from "react";
import "@testing-library/jest-dom";
import axios from "axios";
import {
	render,
	screen,
	act,
	cleanup,
	waitForElementToBeRemoved,
} from "@testing-library/react";
import { App } from "./app.component";
import { resultMockAPI } from "./__test__/mockAPI";

jest.mock("axios");

afterEach(cleanup);

describe("it should render App", () => {
	it("should render App", async () => {
		const promise = Promise.resolve({
			results: resultMockAPI,
		});
		act(() => {
			const { container } = render(<App />);
			expect(container.firstChild.className).toMatch("App");
			const banner = container.querySelector(".banner");
			expect(banner).toBeInTheDocument();
		});
		await act(() => promise);
	});
	it("should show Loading...", async () => {
		const promise = Promise.resolve({
			data: {
				results: resultMockAPI,
			},
		});
		axios.get.mockImplementation(() => promise);
		render(<App />);
		expect(screen.getByText(/loading .../i)).toBeInTheDocument();
		await waitForElementToBeRemoved(() => screen.getByText(/loading .../i));
	});
	it("should display the result from call to movie db api", async () => {
		const promise = Promise.resolve({
			data: {
				results: resultMockAPI,
			},
		});
		axios.get.mockImplementation(() => promise);
		render(<App />);
		expect(screen.getByText(/loading .../i)).toBeInTheDocument();
		await waitForElementToBeRemoved(() => screen.getByText(/loading .../i));
		expect(await screen.findByText(/Godzilla vs. Kong/)).toBeInTheDocument();
		expect(
			await screen.findByText(/Raya and the Last Dragon/)
		).toBeInTheDocument();
		expect(await screen.findByText(/Tom & Jerry/)).toBeInTheDocument();
	});
});

describe("SNAPSHOT", () => {
	it("Snapshot before fetching data from the movie db API", async () => {
		const promise = Promise.resolve({
			data: {
				results: resultMockAPI,
			},
		});
		axios.get.mockImplementation(() => promise);
		let testedUnit;
		act(() => {
			testedUnit = render(<App />);
		});
		expect(testedUnit).toMatchSnapshot();
		await act(() => promise);
	});
	it("Snapshot after fetching data from the movie db API", async () => {
		const promise = Promise.resolve({
			data: {
				results: resultMockAPI,
			},
		});

		axios.get.mockImplementation(() => promise);
		let testedUnit;
		act(() => {
			testedUnit = render(<App />);
		});

		await waitForElementToBeRemoved(() => screen.getByText(/loading .../i));
		expect(testedUnit).toMatchSnapshot();
		await act(() => promise);
	});
});
