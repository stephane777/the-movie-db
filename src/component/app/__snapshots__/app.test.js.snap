// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT Snapshot after fetching data from the movie db API 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <div
            class="banner"
          >
            <div
              class="sc-gKckTs eSiOWf"
            >
              <div
                class="sc-iCfLBT kvZXnL"
              />
            </div>
          </div>
          
          <div
            class="px-0 container"
            style="max-width: 1600px;"
          >
            <div
              class="my-5 mx-0 row"
            >
              <h2
                class="sc-crHlIS eLSBAS"
              >
                Now Playing Movies
              </h2>
            </div>
            <div
              class="d-flex justify-content-center mx-0 row"
            >
              <div
                class="d-flex flex-column align-items-center col-md-4"
              >
                <div
                  class="sc-hKwCoD fmXqTB mt-3 accordion"
                >
                  <div
                    class="card"
                  >
                    <div
                      class="p-0 card-header"
                    >
                      <button
                        class="sc-dkPtyc bQeeSs"
                        type="button"
                      >
                        Sort
                        <div
                          class="sc-jRQAMF fyWpwA"
                        />
                      </button>
                    </div>
                    <div
                      class="collapse"
                      data-testid="filter_list"
                    >
                      <div
                        class="card-body"
                      >
                        <form
                          class=""
                        >
                          <div
                            class="form-group"
                          >
                            <label
                              class="form-label"
                              for="filter.sort"
                            >
                              Sort Results By
                            </label>
                            <select
                              class="sc-eCImvq fNunTe"
                            >
                              <option>
                                Ascending
                              </option>
                              <option>
                                Descending
                              </option>
                            </select>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
                <button
                  class="sc-egiSv fxyaxp btn btn-primary"
                  disabled=""
                  type="button"
                >
                  Search
                </button>
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="sc-bqiQRQ kjxIZB justify-content-start row"
                >
                  <div
                    class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
                  >
                    <img
                      alt="Godzilla vs. Kong"
                      class="card-img-top"
                      src="https://image.tmdb.org/t/p/w500/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
                    />
                    <div
                      class="sc-bdvvaa fzHxCj"
                    >
                      <canvas
                        height="60"
                        id="399566"
                        style="background-color: transparent;"
                        width="60"
                      >
                        85%
                      </canvas>
                    </div>
                    <div
                      class="py-0 card-body"
                      style="transform: translateY(-25px);"
                    >
                      <div
                        class="mb-0 card-title h5"
                      >
                        <strong
                          style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
                        >
                          Godzilla vs. Kong
                        </strong>
                      </div>
                      <p
                        class="card-text"
                      >
                        24 Wed , 2021
                      </p>
                    </div>
                  </div>
                  <div
                    class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
                  >
                    <img
                      alt="Raya and the Last Dragon"
                      class="card-img-top"
                      src="https://image.tmdb.org/t/p/w500/lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg"
                    />
                    <div
                      class="sc-bdvvaa fzHxCj"
                    >
                      <canvas
                        height="60"
                        id="527774"
                        style="background-color: transparent;"
                        width="60"
                      >
                        83%
                      </canvas>
                    </div>
                    <div
                      class="py-0 card-body"
                      style="transform: translateY(-25px);"
                    >
                      <div
                        class="mb-0 card-title h5"
                      >
                        <strong
                          style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
                        >
                          Raya and the Last Dragon
                        </strong>
                      </div>
                      <p
                        class="card-text"
                      >
                        3 Wed , 2021
                      </p>
                    </div>
                  </div>
                  <div
                    class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
                  >
                    <img
                      alt="Tom & Jerry"
                      class="card-img-top"
                      src="https://image.tmdb.org/t/p/w500/6KErczPBROQty7QoIsaa6wJYXZi.jpg"
                    />
                    <div
                      class="sc-bdvvaa fzHxCj"
                    >
                      <canvas
                        height="60"
                        id="587807"
                        style="background-color: transparent;"
                        width="60"
                      >
                        74%
                      </canvas>
                    </div>
                    <div
                      class="py-0 card-body"
                      style="transform: translateY(-25px);"
                    >
                      <div
                        class="mb-0 card-title h5"
                      >
                        <strong
                          style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
                        >
                          Tom & Jerry
                        </strong>
                      </div>
                      <p
                        class="card-text"
                      >
                        11 Thu , 2021
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-furvIG fGdDLR footer"
          >
            <footer
              class="sc-pVTma cgLCRp"
            >
              <a
                class="footer__link"
                href="https://github.com/stephane777"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="sc-kDThTU kQYHNh"
                >
                  <use
                    class="grid__icon-use"
                    href="svg#icon-github"
                  />
                </svg>
              </a>
              <p
                class="sc-jrQzUz ejAwAE"
              >
                Created by Stephane Candelas.
              </p>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        <div
          class="banner"
        >
          <div
            class="sc-gKckTs eSiOWf"
          >
            <div
              class="sc-iCfLBT kvZXnL"
            />
          </div>
        </div>
        
        <div
          class="px-0 container"
          style="max-width: 1600px;"
        >
          <div
            class="my-5 mx-0 row"
          >
            <h2
              class="sc-crHlIS eLSBAS"
            >
              Now Playing Movies
            </h2>
          </div>
          <div
            class="d-flex justify-content-center mx-0 row"
          >
            <div
              class="d-flex flex-column align-items-center col-md-4"
            >
              <div
                class="sc-hKwCoD fmXqTB mt-3 accordion"
              >
                <div
                  class="card"
                >
                  <div
                    class="p-0 card-header"
                  >
                    <button
                      class="sc-dkPtyc bQeeSs"
                      type="button"
                    >
                      Sort
                      <div
                        class="sc-jRQAMF fyWpwA"
                      />
                    </button>
                  </div>
                  <div
                    class="collapse"
                    data-testid="filter_list"
                  >
                    <div
                      class="card-body"
                    >
                      <form
                        class=""
                      >
                        <div
                          class="form-group"
                        >
                          <label
                            class="form-label"
                            for="filter.sort"
                          >
                            Sort Results By
                          </label>
                          <select
                            class="sc-eCImvq fNunTe"
                          >
                            <option>
                              Ascending
                            </option>
                            <option>
                              Descending
                            </option>
                          </select>
                        </div>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="sc-egiSv fxyaxp btn btn-primary"
                disabled=""
                type="button"
              >
                Search
              </button>
            </div>
            <div
              class="col-md-8"
            >
              <div
                class="sc-bqiQRQ kjxIZB justify-content-start row"
              >
                <div
                  class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
                >
                  <img
                    alt="Godzilla vs. Kong"
                    class="card-img-top"
                    src="https://image.tmdb.org/t/p/w500/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
                  />
                  <div
                    class="sc-bdvvaa fzHxCj"
                  >
                    <canvas
                      height="60"
                      id="399566"
                      style="background-color: transparent;"
                      width="60"
                    >
                      85%
                    </canvas>
                  </div>
                  <div
                    class="py-0 card-body"
                    style="transform: translateY(-25px);"
                  >
                    <div
                      class="mb-0 card-title h5"
                    >
                      <strong
                        style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
                      >
                        Godzilla vs. Kong
                      </strong>
                    </div>
                    <p
                      class="card-text"
                    >
                      24 Wed , 2021
                    </p>
                  </div>
                </div>
                <div
                  class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
                >
                  <img
                    alt="Raya and the Last Dragon"
                    class="card-img-top"
                    src="https://image.tmdb.org/t/p/w500/lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg"
                  />
                  <div
                    class="sc-bdvvaa fzHxCj"
                  >
                    <canvas
                      height="60"
                      id="527774"
                      style="background-color: transparent;"
                      width="60"
                    >
                      83%
                    </canvas>
                  </div>
                  <div
                    class="py-0 card-body"
                    style="transform: translateY(-25px);"
                  >
                    <div
                      class="mb-0 card-title h5"
                    >
                      <strong
                        style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
                      >
                        Raya and the Last Dragon
                      </strong>
                    </div>
                    <p
                      class="card-text"
                    >
                      3 Wed , 2021
                    </p>
                  </div>
                </div>
                <div
                  class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
                >
                  <img
                    alt="Tom & Jerry"
                    class="card-img-top"
                    src="https://image.tmdb.org/t/p/w500/6KErczPBROQty7QoIsaa6wJYXZi.jpg"
                  />
                  <div
                    class="sc-bdvvaa fzHxCj"
                  >
                    <canvas
                      height="60"
                      id="587807"
                      style="background-color: transparent;"
                      width="60"
                    >
                      74%
                    </canvas>
                  </div>
                  <div
                    class="py-0 card-body"
                    style="transform: translateY(-25px);"
                  >
                    <div
                      class="mb-0 card-title h5"
                    >
                      <strong
                        style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
                      >
                        Tom & Jerry
                      </strong>
                    </div>
                    <p
                      class="card-text"
                    >
                      11 Thu , 2021
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-furvIG fGdDLR footer"
        >
          <footer
            class="sc-pVTma cgLCRp"
          >
            <a
              class="footer__link"
              href="https://github.com/stephane777"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="sc-kDThTU kQYHNh"
              >
                <use
                  class="grid__icon-use"
                  href="svg#icon-github"
                />
              </svg>
            </a>
            <p
              class="sc-jrQzUz ejAwAE"
            >
              Created by Stephane Candelas.
            </p>
          </footer>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SNAPSHOT Snapshot before fetching data from the movie db API 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <div
            class="banner"
          >
            <div
              class="sc-gKckTs eSiOWf"
            >
              <div
                class="sc-iCfLBT kvZXnL"
              />
            </div>
          </div>
          <div
            class="sc-iqsfdx hJUCHA"
          >
            <div
              class="spinner-border"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading ...
              </span>
            </div>
          </div>
          
          <div
            class="sc-furvIG fGdDLR footer"
          >
            <footer
              class="sc-pVTma cgLCRp"
            >
              <a
                class="footer__link"
                href="https://github.com/stephane777"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="sc-kDThTU kQYHNh"
                >
                  <use
                    class="grid__icon-use"
                    href="svg#icon-github"
                  />
                </svg>
              </a>
              <p
                class="sc-jrQzUz ejAwAE"
              >
                Created by Stephane Candelas.
              </p>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        <div
          class="banner"
        >
          <div
            class="sc-gKckTs eSiOWf"
          >
            <div
              class="sc-iCfLBT kvZXnL"
            />
          </div>
        </div>
        <div
          class="sc-iqsfdx hJUCHA"
        >
          <div
            class="spinner-border"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading ...
            </span>
          </div>
        </div>
        
        <div
          class="sc-furvIG fGdDLR footer"
        >
          <footer
            class="sc-pVTma cgLCRp"
          >
            <a
              class="footer__link"
              href="https://github.com/stephane777"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="sc-kDThTU kQYHNh"
              >
                <use
                  class="grid__icon-use"
                  href="svg#icon-github"
                />
              </svg>
            </a>
            <p
              class="sc-jrQzUz ejAwAE"
            >
              Created by Stephane Candelas.
            </p>
          </footer>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
